cpp/include/cudf/column/column.hpp

```diff
@@ -75,6 +76,30 @@ class column {
    */
   column(column&& other) noexcept;
 
+  /**
+   * @brief Construct a new column by taking ownership of the contents of a device_uvector.
+   *
+   * @param other The device_uvector whose contents will be moved into the new column.
+   * @param null_mask Optional, column's null value indicator bitmask. May
+   * be empty if `null_count` is 0 or `UNKNOWN_NULL_COUNT`.
+   * @param null_count Optional, the count of null elements. If unknown, specify
+   * `UNKNOWN_NULL_COUNT` to indicate that the null count should be computed on
+   * the first invocation of `null_count()`.
+   */
+  template <typename T, CUDF_ENABLE_IF(cudf::is_numeric<T>() or cudf::is_chrono<T>())>
+  column(rmm::device_uvector<T>&& other,
+         rmm::device_buffer&& null_mask = {},
+         size_type null_count           = UNKNOWN_NULL_COUNT)
+  {
+    CUDF_EXPECTS(other.size() <= static_cast<std::size_t>(std::numeric_limits<size_type>::max()),
+                 "The device_uvector size exceeds the maximum size_type.");
+    _type       = cudf::type_to_id<T>();
+    _size       = static_cast<size_type>(other.size());
+    _data       = other.release();
+    _null_mask  = std::forward<rmm::device_buffer>(null_mask);
+    _null_count = null_count;
+  }
```

Comment:
Don't initialize in the ctor body whenever possible. Use an IILE to put error checking in the member init list. 

```suggestion
  column(rmm::device_uvector<T>&& other,
         rmm::device_buffer&& null_mask = {},
         size_type null_count           = UNKNOWN_NULL_COUNT) 
         : _type{cudf::type_to_id<T>()}, 
           _size{ [&](){
              CUDF_EXPECTS(other.size() <= static_cast<std::size_t>(std::numeric_limits<size_type>::max()),
              "The device_uvector size exceeds the maximum size_type.");
               return other.size();
             }() },
             _data{other.release()},
             _null_mask{std::move(null_mask)},
             _null_count{null_count}
  {}
```

Reply:
@jrhemstad I had no idea this was possible! Thank you for the suggestion.

Reply:
Same for me. I thought that when the object was not fully constructed (its constructor doesn't finish) then we can't capture by ref.